protocol_grpc.go

```diff
@@ -421,7 +428,13 @@ func (r *grpcWebClientReceiver) Receive(message any) error {
 	}
 	// See if the server sent an explicit error in the gRPC-Web trailers.
 	mergeHeaders(r.trailer, r.unmarshaler.WebTrailer())
-	if serverErr := grpcErrorFromTrailer(r.bufferPool, r.protobuf, r.trailer); serverErr != nil {
+	serverErr := grpcErrorFromHeaders(r.bufferPool, r.protobuf, r.header, r.trailer)
```

Comment:
We've already checked `r.header` in `grpcValidateResponse`. Do we need to check it again, or can we pass `nil` here?